tools/cldr-code/src/main/java/org/unicode/cldr/util/GrammarInfo.java

```diff
@@ -470,4 +476,88 @@ private void getSourceCaseAndPluralPolish(String gender, String value,
             return;
         }
     }
+
+    /**
+     * Internal class for thread-safety
+     */
+    static class GrammarLocales {
+        static final Set<String> data = CLDRConfig.getInstance().getSupplementalDataInfo().getLocalesWithFeatures(GrammaticalTarget.nominal, GrammaticalScope.units, GrammaticalFeature.grammaticalCase);
+    }
+
+    /**
+     * Return the locales that have either case or gender info for units (or both).
+     */
+    public static Set<String> getGrammarLocales() {
+        return GrammarLocales.data;
+    }
+
+    static final Set<String> INCLUDE_OTHER = ImmutableSet.of(
```

Comment:
should be data driven

Reply:
I thought of that, but wanted to work out exactly the best way to do it.